pinot-spi/src/main/java/org/apache/pinot/spi/config/table/UpsertConfig.java

```diff
@@ -77,6 +77,10 @@ public UpsertConfig(@JsonProperty(value = "mode", required = true) Mode mode,
     _hashFunction = hashFunction == null ? HashFunction.NONE : hashFunction;
   }
 
+  public UpsertConfig(Mode mode) {
```

Comment:
Let's use an empty constructor which is required for the json deserialization

Reply:
gotcha, thanks!